development/extensions/skeleton_extension.rst

```diff
@@ -0,0 +1,750 @@
+    sites.
+
+Requirements
+------------
+
+-  A phpBB board, version 3.1.4 or newer or 3.2.0-b3 or newer.
+-  PHP version 5.3.9 or newer.
+-  PHP ZipArchive module enabled.
+
+Installation
+------------
+
+-  `Download the latest
+   release <https://www.phpbb.com/customise/db/official_tool/ext_skeleton/>`__.
+-  Unzip and install it to your phpBB ``ext/`` folder, e.g.:
+   ``phpBB/ext/phpbb/skeleton``
+-  In phpBB, go to "ACP" > "Customise" > "Extensions" and enable "phpBB
+   Skeleton Extension".
+
+Create an extension
+===================
+
+.. note::
+
+    The Skeleton Extension is based on the Acme Demo extension. As a
+    result, some of its files will have working classes and methods
+    already. These are provided to make the skeleton both functional and
+    educational. You will want to remove any Acme Demo code you do not
+    need from the skeleton and replace it with your own custom code.
+
+phpBB Web interface
+-------------------
+
+The easiest way to use the Skeleton Extension is from its web-based user
+interface (UI).
+
+.. image:: images/skeleton-web-ui.png
+    :width: 200px
+    :align: center
+    :height: 100px
+    :alt: alternate text
+
+In order to create an extension via the web UI, open the phpBB board you
+installed this extension in and visit the "Create skeleton extension" link in the
+board's navigation bar.
+
+You will be presented with a form where you can enter information about
+the extension you want to create. Complete the form and submit it.
+
+A packaged ZIP file will begin to download in your browser.
+Additionally, a copy of it can be found in ``phpBB/store/tmp-ext/``.
+
+Unzip your skeleton extension and begin writing your own code.
+
+Command line interface
+----------------------
+
+The Skeleton Extension can also accommodate users that prefer working
+from a Command Line Interface (CLI).
+
+In order to create an extension via the CLI, you need to open a console
+on your server and ``cd`` to the root directory of the phpBB board where
+you installed this extension:
+
+.. code:: bash
+
+    $ cd ./path/to/phpBB
+
+To create an extension, run:
+
+.. code:: bash
+
+    $ ./bin/phpbbcli.php extension:create
+
+You will be prompted to enter information about the extension you want
+to create.
+
+A packaged ZIP file will be saved to ``phpBB/store/tmp-ext/``.
+
+Unzip your skeleton extension and begin writing your own code.
+
+Documentation
+=============
+
+The following documentation describes how to use the Skeleton Extension
+to configure and generate a skeleton that will suit your needs, so you
+can jump straight into writing your own code.
+
+Extension packaging
+-------------------
+
+Extensions are packaged in a directory structure as follows: ``vendorname/extensionname``.
+For example, the Acme Demo extension uses a packaging structure that looks like ``acme/demo``.
+
+Vendor name
+^^^^^^^^^^^
+
+The vendor name can be your username or any other name you choose to associate 
```

Comment:
I think we can add a check so `phpbb` and `core` are rejected by the skeleton too.